### PR TITLE
KTOR-326 Allow to set followRedirect property for js client engine

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-326](https://youtrack.jetbrains.com/issue/KTOR-326/JS-client-Cannot-change-redirect-policy-by-followRedirectsfalse)

### CODE REVIEW

File: ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt [hunk 1]
```diff
@@ -31,13 +33,14 @@ internal class JsClientEngine(override val config: HttpClientEngineConfig) : Htt
     @OptIn(InternalAPI::class)
     override suspend fun execute(data: HttpRequestData): HttpResponseData {
         val callContext = callContext()
+        val clientConfig = data.attributes[CLIENT_CONFIG]
```
Comment (rsinukov) [inline ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:36]: why pass it through data? there is a property `config` in class

File: ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt [hunk 2]
```diff
@@ -129,7 +134,7 @@ private fun Event.asString(): String = buildString {
     append(JSON.stringify(this@asString, arrayOf("message", "target", "type", "isTrusted")))
 }
 
-private fun io.ktor.client.fetch.Headers.mapToKtor(): Headers = buildHeaders {
+private fun org.w3c.fetch.Headers.mapToKtor(): Headers = buildHeaders {
```
Comment (hfhbd) [inline ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:137]: Are you sure this works on nodeJS? See also https://github.com/ktorio/ktor/pull/2592

Comment (e5l) [reply ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:137]: In this case, we replace some types, and the actual implementation stays the same.

Comment (e5l) [reply ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:137]: The tests are also fine

Comment (hfhbd) [reply ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:137]: Unit tests? AFAIK they does not work with nodeJS production, at least 1 year ago, which results into the same problem at runtime: https://github.com/ktorio/ktor/pull/2385 so you needed to create a test project and use the artifact.

File: ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:36]: cause we need client config, not engine

File: ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:137]: I will try to set up a sample

File: ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt (hunk 1 above)
Comment (rsinukov) [reply ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsClientEngine.kt:36]: Ok. Let's create a ticket to pass client config in client factory `HttpClientEngineFactory.create` method in 3.0.0 for a proper fix.

**Review by rsinukov — APPROVED** [review]
LGTM, but please check sample with node

**Comment by e5l** [comment]
Sure, https://youtrack.jetbrains.com/issue/KTOR-4484/Pass-Client-Config-to-HttpClientEngine

**Pull request merged**